File: design-patterns/ts/visitor/ui-element/head-element.ts
// head-element.ts

import UIElement from "./ui-element";
import Visitor from "./visitor";

class HeadElement implements UIElement {
    private readonly wrapper: string = 'h1';

    constructor(private text: string) {

    }

    getText(): string {
        return this.text
    }

    getWrapper(): string {
        return this.wrapper;
    }

    appendElement(visitor: Visitor): void {
        visitor.appendContentWithWrapper(this);
    }
}

export default HeadElement;

File: design-patterns/go/visitor/ui_element/head_element.go
// head_element.go

package main

type HeadElement struct {
	wrapper string
	text    string
}

func NewHeadElement(text string) (headElement *HeadElement) {
	headElement = &HeadElement{}
	headElement.wrapper = "h1"
	headElement.text = text
	return
}

func (headElement *HeadElement) GetText() string {
	return headElement.text
}

func (headElement *HeadElement) GetWrapper() string {
	return headElement.wrapper
}

func (headElement *HeadElement) AppendElement(visitor Visitor) {
	visitor.AppendContentWithWrapper(headElement)
}
